src/nvim/msgpack_rpc/channel.c

```diff
@@ -360,7 +360,11 @@ static void handle_request(Channel *channel, msgpack_object *request)
       request_event((void **)&evdata);
     }
   } else {
-    multiqueue_put(channel->events, request_event, 1, evdata);
+    if (!is_remote_client) {
```

Comment:
we should probably set `async=true` on the "redraw" handler instead (if not already, then this can just be reverted).